scene/main/canvas_item.h

```diff
@@ -427,4 +427,6 @@ class CanvasTexture : public Texture2D {
 	~CanvasTexture();
 };
 
+#define ERR_DRAW_GUARD ERR_FAIL_COND_MSG(!drawing, "Drawing is only allowed inside this node's `_draw()`, functions connected to its `draw` signal, or when it receives NOTIFICATION_DRAW.");
```

Comment:
Since it's only used in `canvas_item.cpp`, it should be moved there.

Otherwise it's going to be available to anything that includes `canvas_item.h` (like 2/3rd the nodes and all editor code), while it's not actually usable there since `drawing` is a member of `CanvasItem`.

Reply:
Also, nitpicky but I would style it like this:
```suggestion
#define ERR_DRAW_GUARD \
		ERR_FAIL_COND_MSG(!drawing, "Drawing is only allowed inside this node's `_draw()`, functions connected to its `draw` signal, or when it receives NOTIFICATION_DRAW.")
```
And note I removed the semicolon - this would make sure `ERR_DRAW_GUARD` needs to be used with a semicolon to look like a method / be consistent. And avoid otherwise getting two semicolons in a row after the pre-processor does its magic.